src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -61,6 +61,25 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 			this._pwshAddon?.dispose();
 		}));
 		this._terminalSuggestWidgetVisibleContextKey = TerminalContextKeys.suggestWidgetVisible.bindTo(this._contextKeyService);
+		this.add(this._configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(TerminalSuggestSettingId.EnableExtensionCompletions) || e.affectsConfiguration(TerminalSuggestSettingId.Enabled)) {
+				const extensionCompletionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enableExtensionCompletions;
+				const completionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enabled;
+				if (!extensionCompletionsEnabled) {
```

Comment:
Could we make this `!extensionCompletionsEnabled || !completionEnabled` instead and remove it from the lower if?

Reply:
makes it a little more confusing imo...

Reply:
oh, I see what you mean, fixed